2199/2/versus.rb

```diff
@@ -0,0 +1,141 @@
+require 'yaml'
+require 'bundler'
+Bundler.require
+
+# :reek:InstanceVariableAssumption
+# class Battle to collect information
+class Battle
+  def initialize(filename)
+    @filename = filename
+  end
+
+  def bad_words_count
+    words.count do |word|
+      word.include?('*') || RussianObscenity.obscene?(word)
+    end
+  end
+
+  def words_count
+    words.count
+  end
+
+  def rounds_count
+    rounds_count = text.scan(/Раунд \d/).length
+    rounds_count.zero? ? 1 : rounds_count
+  end
+
+  private
+
+  def text
+    @text ||= File.read(@filename)
+  end
+
+  def words
+    return @words if @words
+    @words = text.scan(/[\wа-яА-ЯёЁ*]+/)
+    @words.delete('***')
+    @words
+  end
+end
+
+# Participant container
+class Participant
+  attr_reader :name
+
+  def initialize(name)
+    @name = name
+    @battles = []
+  end
+
+  def add_battle(battle)
+    @bad_words_count = nil
```

Comment:
This is a hack. Right now we nullify only one variable but take a look at `#words_count`. This may also require a memoization.

State mutation is evil. Let's pass all the battles to initializer and never change them.

Or you can return a new instance of `Participant` each time somebody adds a battle leaving the old instance immutable.